subsys/mgmt/CMakeLists.txt

```diff
@@ -1,14 +1,3 @@
 # SPDX-License-Identifier: Apache-2.0
 
-zephyr_library()
-zephyr_library_sources(buf.c)
-zephyr_library_sources(smp.c)
-zephyr_library_sources_ifdef(CONFIG_MCUMGR_SMP_BT smp_bt.c)
-zephyr_library_sources_ifdef(CONFIG_MCUMGR_SMP_SHELL smp_shell.c)
-zephyr_library_sources_ifdef(CONFIG_MCUMGR_SMP_UART smp_uart.c)
-zephyr_library_sources_ifdef(CONFIG_MCUMGR_SMP_UDP smp_udp.c)
-zephyr_library_link_libraries(MCUMGR)
-
-if (CONFIG_MCUMGR_SMP_SHELL OR CONFIG_MCUMGR_SMP_UART)
-    zephyr_library_sources(serial_util.c)
-endif ()
+add_subdirectory_ifdef(CONFIG_MCUMGR               mcumgr)
```

Comment:
This could be probably replaced with `add_subdirectory_if_kconfig`

Reply:
Good point.

Reply:
@tejlmand should we deprecate the `if_kconfig()` in favor of  `_ifdef` ?

Reply:
@carlescufi Currently there are 42 instances of `add_subdirectory_if_kconfig` in CMakeLists.txt of Zephyr. I think that we could RFC the change. I have looked into the macro and thought about it a little and I think that it is not needed as it is just specialized `_ifdef`. When change to the config identifier happens or the file name gets changed you may end up replacing the macro, in CMakeLists.txt` with the `_ifdef` variant anyway.

Reply:
Agreed, I think we should remove it altogether and replace all instances with `_ifdef`. We make no promises of backwards compatibility in CMake macros or functions, so there's no need for deprecation here.

Reply:
#27290